src/test_workflow/compare_benchmark/compare_test_runner.py

```diff
@@ -0,0 +1,62 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+import os
+import subprocess
+
+from test_workflow.compare_benchmark.compare_args import CompareArgs
+
+
+class CompareTestRunner:
+    def __init__(self) -> None:
+        self.container_name = "compare-container"
+
+    def run_comparison(self, compare_args: CompareArgs) -> None:
+        """
+        Perform the comparison between two test results using the OSB tool.
+        """
+
+        # construct the command to compare two test executions
+        command = (
+            f"docker run --name {self.container_name} "
+            "-v ~/.benchmark/benchmark.ini:/opensearch-benchmark/.benchmark/benchmark.ini "
+            f"opensearchproject/opensearch-benchmark:1.6.0 "
+            f"compare --baseline={compare_args.id1} --contender={compare_args.id2} "
+        )
+
+        if compare_args.results_format:
+            command += f"--results-format={compare_args.results_format} "
+
+        if compare_args.results_numbers_align:
+            command += f"--results-numbers-align={compare_args.results_numbers_align} "
+
+        if compare_args.results_file:
+            # create a temporary directory for the compare results file in the container
+            container_results_dir = "/tmp/results"
```

Comment:
The final results file is actually written inside the container which then should be copied to local once the process has completed. 
this argument should just be `final_results.md` so that the file is created inside the container at `/opensearch-benchmark/` path. 
Same is happening with nightly benchmark runs, see https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/benchmark_test/benchmark_test_suite.py#L80. 

The file is then copied to local, see https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/benchmark_test/benchmark_test_suite.py#L96-L101